models/migrations/migrations.go

```diff
@@ -319,6 +320,167 @@ Please try upgrading to a lower version first (suggested v1.6.4), then upgrade t
 	return nil
 }
 
+// RecreateTables will recreate the tables for the provided beans using the newly provided bean definition and move all data to that new table
+// WARNING: YOU MUST PROVIDE THE FULL BEAN DEFINITION
+func RecreateTables(beans ...interface{}) func(*xorm.Engine) error {
+	return func(x *xorm.Engine) error {
+		sess := x.NewSession()
+		defer sess.Close()
+		if err := sess.Begin(); err != nil {
+			return err
+		}
+		for _, bean := range beans {
+			log.Info("Recreating Table: %s for Bean: %s", x.TableName(bean), reflect.Indirect(reflect.ValueOf(bean)).Type().Name())
+			if err := recreateTable(sess, bean); err != nil {
+				return err
+			}
+		}
+		return sess.Commit()
+	}
+}
+
+// recreateTable will recreate the table using the newly provided bean definition and move all data to that new table
+// WARNING: YOU MUST PROVIDE THE FULL BEAN DEFINITION
+// WARNING: YOU MUST COMMIT THE SESSION AT THE END
+func recreateTable(sess *xorm.Session, bean interface{}) error {
+	// TODO: This will not work if there are foreign keys
+
+	tableName := sess.Engine().TableName(bean)
+	tempTableName := fmt.Sprintf("tempzzz__%s__zztemp", tableName)
```

Comment:
Maybe something more obvious like `recreate_tmp_%s`?

Reply:
it's a temporary table that you do not want to exist at the end of the period - anything that makes it more obvious makes it more likely that there is a potential conflict. No-one would ever deliberately name their tables tempzzz__%s__zztemp. recreate_tmp_%s does however stand that chance.

Reply:
however I'll add the prefix `recreate_` - but given how long and difficult getting this to work for MSSQL has been it's highly likely that it will now find something to complain about that.

Reply:
I'm more concerned if the script fails, it may leave that temp table around and with a name like `tempzzz__` no one would guess what it relates to.

Reply:
It all happens within a transaction - if the script fails those temporary tables will get rollbacked. It seems however that MSSQL has tolerated the adjusted name.